htcrystalball/collect.py

```diff
@@ -1,10 +1,10 @@
 """Retrieve, format, and store a system's condor slot configuration."""
 
-from . import SLOTS_CONFIGURATION
-from .utils import kib_to_gib, mib_to_gib
-from typing import Union
 import json
 import htcondor
+from typing import Union
+from . import SLOTS_CONFIGURATION
+from .utils import kib_to_gib, mib_to_gib
```

Comment:
Personally I prefer to put local application/library specific imports on top, but if we're going to listen to pep8 that strictly, you'd have to do the following (see [pep8](https://www.python.org/dev/peps/pep-0008/#imports)).

The order of imports is:
- Standard library imports (e.g. typing, os, json)
- Related third party imports (e.g. htcondor)
- Local application/library specific imports (everything starting with a . ;))

A blank line needs to be put between each of the aforementioned import groups.

Although it seems as if pylint is suggesting to put the `from x import y` under `import`, I cannot find any pep8 guideline on this (although it is safer to assume pylint is correct rather than me ;))

Lastly, while relative imports (e.g. `from . import SLOT_CONFIGURATION`) aren't disallowed, pep8 suggests a preference for absolute imports (e.g. `from htcrystalball import SLOT_CONFIGURATION`). Both should work.

Reply:
> I prefer to put local application/library specific imports on top, but if we're going to listen to pep8 that strictly, you'd have to do the following [...]

We should definitely follow PEP8 here. That is the widespread convention.

> Although it seems as if pylint is suggesting to put the from x import y under import, I cannot find any pep8 guideline on this (although it is safer to assume pylint is correct rather than me ;))

I'm not familiar with the rule, but pylint should provide a rule ID, which you can lookup. Usually there's docs and/or long discussion to be found around those rules, often in the original PR into pylint that introduced the feature.

Reply:
@jonamfischer can you check the rule ID that pylint provides telling you to list `from x import y` below `import x.y`? Also, PEP8 wants the aforementioned import groups to be separated by a blank line, so it should look like:

```
import json

import htcondor

from typing import Union

from . import SLOTS_CONFIGURATION
from .utils import kib_to_gib, mib_to_gib
```

This applies to the other import changes as well.

Reply:
@jonamfischer I forgot to add that since you're changing the imports, it's probably better to change the relative imports to absolute imports, resulting in:

```
import json

import htcondor

from typing import Union

from htcrystalball import SLOTS_CONFIGURATION
from htcrystalball.utils import kib_to_gib, mib_to_gib
```